### PR TITLE
add API to generate leaf exception from exceptongroup

### PR DESCRIPTION
**Pull request by WindSoilder**
try to resolve for issue #6

### CODE REVIEW

**Comment by njsmith** [comment]
Looks sensible to me. I'm not wild about the `leaf_exceptions` name, but I'm not thinking of anything better either. @oremanj and @1st1, you've both had opinions about things adjacent to this in the past, any thoughts? (You'll probably want to look at #6 for the description of what this is for.)

File: exceptiongroup/_tools.py [hunk 1]
```diff
@@ -6,6 +6,25 @@
 from . import ExceptionGroup
 
 
+def leaf_exceptions(exc):
+    """ Iterate and extract sub exceptions from given exception.
```
Comment (oremanj) [inline exceptiongroup/_tools.py]: Can you expand upon this a little bit? I don't think the current docstring would be clear to someone who's not already super familiar with ExceptionGroup. Maybe some examples would help clarify?

File: exceptiongroup/_tools.py [hunk 2]
```diff
@@ -6,6 +6,25 @@
...
+
+    Args:
+        exc (BaseException): Exception object we want to extract.
+    Yields:
+        leaf exception object.
```
Comment (oremanj) [inline exceptiongroup/_tools.py]: Usually we would write this as "Returns: an iterable of [...]"; to me "Yields" seems to be tied to a specific implementation (a generator) which we might not choose to keep in the future.

**Review by oremanj — COMMENTED** [review]
I'm also not wild about the name, and think the implementation looks fine. Considering that people will typically invoke this as `for subexc in ExceptionGroup.<thing>(exc):`, we don't necessarily need a long name. Some ideas that come to mind are `expand`, `flatten`, `iterate` (or maybe `iter`).

**Comment by WindSoilder** [comment]
What about `walk` (the word comes from `os.walk`)?
And I like `iterate` :)

**Pull request closed without merging**